### ISSUE TITLE
MD055 ignores configured table pipe style and falls back to leading-and-trailing

### ISSUE THREAD

**Issue by hal1278** [rvben/rumdl#611]
### rumdl version

0.1.88

### Installation method

Built from source

### Bug description

Some MD055 style values appear to be accepted by configuration parsing, but then fall back to `leading-and-trailing` during rule execution.

I reproduced this with MD055 style values that contain hyphens or underscores, except for `leading-and-trailing` / `leading_and_trailing`, where the fallback is indistinguishable from the requested style.

For example, when `MD055.style` is configured as `no-leading-or-trailing`, a table that already has no leading or trailing pipes is still reported as violating MD055. `rumdl check` reports that the table pipe style should be `leading and trailing`.

The same behavior is reproducible with the snake_case variant `no_leading_or_trailing`, and with other non-default style values such as `leading-only` / `leading_only` and `trailing-only` / `trailing_only`.

This does not look like a configuration discovery issue, because `rumdl config --config ... get MD055.style` reports that the value is read from project config.

### Steps to reproduce

1. Save the configuration below as `.rumdl.toml`.

2. Create a Markdown file that already uses no leading or trailing table pipes:

```sh
cat >repro.md <<'EOF'
# repro.md

A | B
--- | ---
1 | 2
EOF
```

3. Run:

```sh
rumdl config --config .rumdl.toml get MD055.style
rumdl check --no-cache --config .rumdl.toml repro.md
```

4. Then change only `MD055.style` to the snake_case variant and run the same commands again:

```toml
style = "no_leading_or_trailing"
```

I also reproduced the same fallback behavior with `leading-only` / `leading_only` and `trailing-only` / `trailing_only` by using corresponding Markdown tables for each expected style.

### Expected behavior

MD055 should enforce the configured style value.

For example, when `MD055.style` is `no-leading-or-trailing` or `no_leading_or_trailing`, the table in `repro.md` should not report an MD055 issue because it already has no leading or trailing pipes.

In the minimal reproduction above, `rumdl check ...` should complete without MD055 warnings.

Similarly, `leading-only` / `leading_only` and `trailing-only` / `trailing_only` should enforce their respective configured styles instead of falling back to `leading-and-trailing`.

### Actual behavior

For `style = "no-leading-or-trailing"`, `rumdl config` reports that the project config value is read:

```
MD055.style = "no-leading-or-trailing" [from project config]
```

But `rumdl check` reports MD055 warnings requiring leading and trailing pipes:

```
repro.md:3:1: [MD055] Table pipe style should be leading and trailing [*]
repro.md:4:1: [MD055] Table pipe style should be leading and trailing [*]
repro.md:5:1: [MD055] Table pipe style should be leading and trailing [*]
```

The same behavior occurs with `style = "no_leading_or_trailing"`.

I also observed that `rumdl fmt` fixes tables toward the leading-and-trailing style for the affected configurations, but `check` is enough to show the configuration/runtime mismatch.

### Configuration

```toml
[global]
cache = false

[MD055]
style = "no-leading-or-trailing"
```

### Operating system

Linux

### Additional context

This appears to be a mismatch between MD055 style normalization and MD055 rule execution:

- `MD055Config::deserialize_style()` normalizes underscores to hyphens, so values such as `no_leading_or_trailing` are stored as `no-leading-or-trailing`.
- `MD055TablePipeStyle::check()` then validates only snake_case runtime values and treats the normalized kebab-case values as invalid, falling back to `leading_and_trailing`.

Therefore, normalized kebab-case style values may be accepted by configuration parsing but then treated as invalid during MD055 rule execution, causing the fallback to `leading-and-trailing`.

As a local verification, a small patch that accepts both kebab-case and snake_case values in `MD055TablePipeStyle::check()` and maps them to the existing snake_case runtime values made all tested MD055 cases behave as expected.

I have only verified this runtime behavior for MD055. I have not checked whether other rules with style normalization have a similar issue.

**Comment by rvben** [comment]
Fixed in v0.1.89. The root cause was a case-normalization mismatch in `MD055Config::deserialize_style()`: it was normalizing config values to kebab-case (`no-leading-or-trailing`), while the rule's internal match arms, `determine_pipe_style()`, and `fix_table_content()` all use snake_case (`no_leading_or_trailing`). Any kebab-case value fell through to the wildcard arm and silently defaulted to `leading_and_trailing`.

The fix normalizes both forms to snake_case on deserialization, so `no-leading-or-trailing`, `no_leading_or_trailing`, and `NO_LEADING_OR_TRAILING` all work identically. Same fix covers `leading-only`, `trailing-only`, and `leading-and-trailing`.

Thanks for the detailed report and the precise root-cause analysis, @hal1278 — it made the fix straightforward.
